### PR TITLE
[feature] Accept custom relative thresholds in duration.humanize

### LINKED ISSUE #4295: Feature Request: Pass relativeTimeThresholds to humanize() as argument
_Note:_ PR #4296 has been submitted for this.

**Description of the Issue and Steps to Reproduce:**

1. Have an application which wants to use different `relativeTimeThreshold` values in different parts of its application
1. Call `relativeTimeThreshold` in one part of the application

**Desired:** There is a way to confine those relative time thresholds to that part of the application
**Observed:** The changes are global

# Proposal
Change the signature of `humanize` as follows:

```typescript
export function humanize(withSuffixOrOptions?: boolean | Options): string;

export interface Options {
    withSuffix?: boolean;
    thresholds?: Thresholds;
}

// Define a type with the current set of keys accepted by relativeTimeThreshold
export type Thresholds = typeof thresholds;
```

Within `humanize`, if `withSuffixOrOptions` is an object and it contains a `thresholds` property, that will be used instead of the `var thresholds` defined earlier in the file.

### PR DESCRIPTION
**Pull request by TedDriggs**
Fixes #4295

### CODE REVIEW

File: src/test/moment/duration.js [hunk 1]
```diff
@@ -464,6 +464,24 @@ test('humanize duration with suffix', function (assert) {
...
+    moment.locale('en');
+    assert.equal(
+        moment.duration({seconds: -10}).humanize({thresholds: thresholds}),
+        'a minute',
+        '10 seconds = a minute (with thresholds)'
+    );
+    assert.equal(
+        moment.duration({seconds: 10}).humanize({thresholds: thresholds, withSuffix: true}),
+        'in a minute',
+        '100 seconds = a minute (with thresholds and suffix)'
```
Comment (TWiStErRob) [inline src/test/moment/duration.js]: `{seconds: 10}` VS "100 seconds`, one of these is wrong.

Comment (TedDriggs) [reply src/test/moment/duration.js]: Looks like the message has an extra zero. I'll put up a fix.

**Comment by ichernev** [comment]
Merged in 757d4ff8e57359681a2625f4413bdf6f81376b0e

**Comment by JaxonWright** [comment]
This allegedly is in 2.25, but I still don't seem to have it after upgrading to 2.25.3.

**Pull request closed without merging**